### ISSUE TITLE
I meet this problem when I run the demo.py. How to solve it?

### ISSUE THREAD

**Issue by yyjFish** [gpleiss/efficient_densenet_pytorch#24]
> 0.1
Training
Traceback (most recent call last):
  File "demo.py", line 272, in <module>
    fire.Fire(demo)
  File "/home/yyj/anaconda2/lib/python2.7/site-packages/fire/core.py", line 127, in Fire
    component_trace = _Fire(component, args, context, name)
  File "/home/yyj/anaconda2/lib/python2.7/site-packages/fire/core.py", line 366, in _Fire
    component, remaining_args)
  File "/home/yyj/anaconda2/lib/python2.7/site-packages/fire/core.py", line 542, in _CallCallable
    result = fn(*varargs, **kwargs)
  File "demo.py", line 250, in demo
    n_epochs=n_epochs, batch_size=batch_size, seed=seed)
  File "demo.py", line 166, in train
    train=True,
  File "demo.py", line 101, in run_epoch
    output_var = model(input_var)
  File "/home/yyj/anaconda2/lib/python2.7/site-packages/torch/nn/modules/module.py", line 325, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/yyj/Downloads/efficient_densenet_pytorch-master/models/densenet_efficient.py", line 218, in forward
    features = self.features(x)
  File "/home/yyj/anaconda2/lib/python2.7/site-packages/torch/nn/modules/module.py", line 325, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/yyj/anaconda2/lib/python2.7/site-packages/torch/nn/modules/container.py", line 67, in forward
    input = module(input)
  File "/home/yyj/anaconda2/lib/python2.7/site-packages/torch/nn/modules/module.py", line 325, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/yyj/Downloads/efficient_densenet_pytorch-master/models/densenet_efficient.py", line 152, in forward
    outputs.append(module.forward(outputs))
  File "/home/yyj/Downloads/efficient_densenet_pytorch-master/models/densenet_efficient.py", line 107, in forward
    new_features = super(_DenseLayer, self).forward(prev_features)
  File "/home/yyj/anaconda2/lib/python2.7/site-packages/torch/nn/modules/container.py", line 67, in forward
    input = module(input)
  File "/home/yyj/anaconda2/lib/python2.7/site-packages/torch/nn/modules/module.py", line 325, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/yyj/Downloads/efficient_densenet_pytorch-master/models/densenet_efficient.py", line 85, in forward
    return fn(self.norm_weight, self.norm_bias, self.conv_weight, *inputs)
  File "/home/yyj/Downloads/efficient_densenet_pytorch-master/models/densenet_efficient.py", line 265, in forward
    conv_output = self.efficient_conv.forward(conv_weight, None, relu_output)
  File "/home/yyj/Downloads/efficient_densenet_pytorch-master/models/densenet_efficient.py", line 457, in forward
    self.groups, cudnn.benchmark
TypeError: _cudnn_convolution_full_forward received an invalid combination of arguments - got (torch.cuda.FloatTensor, torch.cuda.FloatTensor, NoneType, torch.cuda.FloatTensor, tuple, tuple, tuple, int, bool), but expected (torch.cuda.RealTensor input, torch.cuda.RealTensor weight, torch.cuda.RealTensor bias, torch.cuda.RealTensor output, std::vector<int> pad, std::vector<int> stride, std::vector<int> dilation, int groups, bool benchmark, bool deterministic)

**Comment by taineleau** [comment]
Could you please specify your environment (OS, PyTorch version)?

**Comment by Vandermode** [comment]
also meet this error. it seems cudnn wrapper api changed.
```
TypeError: _cudnn_convolution_full_forward received an invalid combination of arguments - got (torch.cuda.FloatTensor, torch.cuda.FloatTensor, NoneType, torch.cuda.FloatTensor, tuple, tuple, tuple, int,bool), but expected (torch.cuda.RealTensor input, torch.cuda.RealTensor weight, torch.cuda.RealTensor bias, torch.cuda.RealTensor output, std::vector<int> pad, std::vector<int> stride, std::vector<int> dilation, int groups, bool benchmark, bool deterministic)
```
```python
import torch
torch.__version__
'0.4.0a0+8ebf18b'
```
OS version is ubuntu 16

**Comment by yyjFish** [comment]
@taineleau  appreciate for your answer.

This is the environment of my computer:

Pytorch  ———— 0.3
Cuda ————  8.0
Cudnn ———— 5.1
OS ———— Linux16.04

**Comment by yyjFish** [comment]
@Vandermode  I'm agree with you,       Do you solve this problem?

**Comment by gpleiss** [comment]
Right now we're not compatible with PyTorch verison 0.3. We'll work on it!
If you don't mind, I'll close this issue and open a more generic "0.3 compatibility issue."

**Comment by taineleau** [comment]
Some updates.
I found the pytorch 0.3 changed the API a little bit. For `_cudnn_convolution_*`, you have to pass a `deterministic` flag.  see [this branch](https://github.com/gpleiss/efficient_densenet_pytorch/tree/pytorch0.3). This branch is runnable under 0.3., but the result is wrong.

**Comment by MattKleinsmith** [comment]
Some test results added to: https://discuss.pytorch.org/t/0-3-cudnn-convolution-deterministic-flag/11617/11?u=matthewkleinsmith

**Comment by gpleiss** [comment]
I'm going to try a fresh implementation of this, which doesn't use the low-level api calls. What I'm thinking is using the functional api, and then being clever with variable detaching and the such.

I'll hopefully have something to post this weekend!

**Comment by gpleiss** [comment]
Should be fixed now with #28.
